### PR TITLE
Drop absent file declarations

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: manifests/ca.pp [hunk 1]
```diff
@@ -23,10 +23,6 @@
 ) {
   $server_ca_path = "${certs::ssl_build_dir}/${server_ca_name}.crt"
 
-  file { "${certs::pki_dir}/private/${default_ca_name}.pwd":
-    ensure => absent,
```
Comment (ekohl) [inline manifests/ca.pp:27]: This was ensured absent in 4ba477a68860d1ce7085699020a83755f066debf, released in 17.0.0.

File: manifests/ca.pp [hunk 2]
```diff
@@ -76,12 +72,6 @@
   }
 
   if $deploy {
-    # Ensure CA key deployed to /etc/pki/katello/private no longer exists
-    # The CA key is not used by anything from this directory and does not need to be deployed
-    file { $ca_key:
-      ensure => absent,
```
Comment (ekohl) [inline manifests/ca.pp:82]: Ensured absent in 028f93af283b718752e6263b2732c8c0ce308caf, released in 14.0.0.

File: manifests/candlepin.pp [hunk 3]
```diff
@@ -28,24 +28,6 @@
   $java_client_cert_name = 'java-client'
   $artemis_alias = 'artemis-client'
   $artemis_client_dn = $certs::foreman::client_dn
-
-  cert { $java_client_cert_name:
-    ensure        => absent,
```
Comment (ekohl) [inline manifests/candlepin.pp:33]: Ensured absent in 0922645cac5bc3c1ce6900badf5d8aba632afa13, released in 13.0.0.

File: manifests/candlepin.pp [hunk 4]
```diff
@@ -89,22 +71,6 @@
       key_decrypt       => true,
     }
 
-    file { "${pki_dir}/private/katello-tomcat.key":
-      ensure  => absent,
```
Comment (ekohl) [inline manifests/candlepin.pp:93]: Ensured absent in 8e0050580196ae4f9132ff395885f481314b52d and, released in 14.0.0, but I think prior to that already.

**Review by ekohl — COMMENTED** [review]
The latest addition was in 17.0.0, which was first included in Foreman 3.9. I'm a bit hesitant about that. The others go back to Foreman 3.1, which included 15.0.0 and I feel safe about that.

I'm tempted to wait a bit longer with `${default_ca_name}.pwd`, but we don't officially support skipping releases so perhaps it's irrational.

@evgeni mind being the deciding third?

**Comment by evgeni** [comment]
I like being conservative:)

**Comment by ekohl** [comment]
@ehelms merged #452 so we already know it won't be part of 18.0.0 :)

**Comment by evgeni** [comment]
Is it really backwards incompatible tho?
No parameters are removed and no change of functionality happens. We just stop managing resources that should be absent.

**Comment by ehelms** [comment]
I think now that 18.0.0 is released, we can merge this?

**Pull request merged**